Rename server_uuids parameter to servers in volume module

ctx opened the pull request.

ctx committed:
Rename server_uuids parameter to servers in volume module

The server_uuid(s) attribute is marked as deprecated on the cloudscale api
description. For now the old names are aliases for the server parameter.

Fixes: #53

href commented on plugins/modules/volume.py:
```diff
@@ -176,6 +176,33 @@ def create(self, resource):
+        if key == 'servers':
+            server_has = resource[key]
+            server_wanted = param
+            if len(param) > len(server_has):
```
Any reason why you check for "greather than" only? I would assume there is a difference whenever there's a difference in length.

href replied:
Also, if you already set `server_wanted` to be equal to `param`, which I think is reasonable, you should refer to it here using `server_wanted` and not `param`.

href commented on plugins/modules/volume.py:
```diff
@@ -176,6 +176,33 @@ def create(self, resource):
+                        is_different = True
+
+        # If it looks like a stub
+        elif isinstance(resource[key], dict) and 'href' in resource[key]:
```
This is the same logic already implemented in the base module. You should defer to the base class using `super()`. If you want you could even check if the base class already determined a difference. If so, you don't need to check.

If there is no difference and the key is 'server', then you could do your check.

href reviewed (changes requested):
There are a few issues with your code that you need to fix, most of it is small, but the code duplication between the base and the volume is a blocker for me, as it'll lead to bugs in the future, when someone misses the second occurrence of the same code.

ctx committed:
Update examples so that the variables are used

gaudenz reviewed (changes requested):
Please add a changelog entry for the parameter rename. We proabably also want to deprecate the `server_uuids` and `server_uuid` variants.

gaudenz committed:
Deprecate aliases for servers in volume module

The aliases `server_uuids` and `server_uuid` are deprecated and will be
removed in version 3.0.0. This follows the deprecation of the `server_uuids`
parameter in the API.

gaudenz commented:
This now looks good from my point of view. But I'd prefer @href to have a quick look at my deprecation commit.

href commented:
Looks good to me, go ahead and merge at will.

resmo commented on plugins/modules/volume.py:
```diff
@@ -145,7 +147,7 @@
   type: dict
   sample: {'slug': 'lpg1'}
-server_uuids:
+servers:
```
hmm server_uuids is still returned as list of keys, but servers is also returned as list of objects from the api or did I miss a thing? So I would just add a new return instead of rename and make a proper sample

gaudenz replied:
You are right. Will fix this. Thanks for spotting this.

resmo commented:
I see a missed opportunity: 

What about using a list of objects as param (similar what is returned by the api)?

Advantages:
* comparison of diff may be easier (compare list of objects)
* server by name (no need to query and register the result for uuid of the server in a previous tasks)

# by server uuid
```yaml
- cloudscale_ch.cloud.volume:
    uuid: "{{ my_ssd_volume.uuid }}"
    servers:
    - uuid: ea3b39a3-77a8-4d0b-881d-0bb00a1e7f48
```
#  by server name
```yaml
- cloudscale_ch.cloud.volume:
    uuid: "{{ my_ssd_volume.uuid }}"
    servers:
    - name: my-server
```

ctx commented:
Thanks for this suggestion, I created an issue. We will find more parameters which could take a name or an uuid and even an object: https://github.com/cloudscale-ch/ansible-collection-cloudscale/issues/56

gaudenz reviewed (approved):
LGTM now. Thanks.
